### PR TITLE
LPS-173661 KB Navigation Search

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
This is the first PR for the search filtering in the navigation panel. All this feature is under FF (feature.flag.LPS-156421)

For now, the search results are shown as (when more than 3 characters have been introduced):

 
Screenshot 2023-01-27 at 12 40 22

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+const ITEM_TYPES_SYMBOL = {
+	article: 'document-text',
+	folder: 'folder',
+};
+
+const SEARCH_DELTA = 2;
+
+export default function SearchField({handleSearchChange, items}) {
+	const initialState = {
```
Comment (jaimelr10) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js:28]: Hi @ambrinchaudhary :)
Do you think we could change some of the names to make them more self-explanatory? 
What do you think about `initialSearchInfo` for this?

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+	const [state, setState] = useState(initialState);
+
+	const [searchActive, setSearchActive] = useState(false);
+
+	useEffect(() => {
+		handleSearchChange({isSearchActive: searchActive});
+	}, [handleSearchChange, searchActive]);
+
+	const handleChange = (event) => {
```
Comment (jaimelr10) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js:41]: Is it ok to change this to `handleSearchQueryChange`?

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		}
+
+		setSearchActive(newQuery.length > SEARCH_DELTA);
+		setState({
+			filteredItems: results,
+			query: newQuery,
+		});
+	};
+
+	const handleClick = () => {
```
Comment (jaimelr10) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js:58]: What do you think of `handleSearch` or `handleSearchClick` for this one?

**Review by jaimelr10 — COMMENTED** [review]
What do you think about this? :)

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js (hunk 3 above)
Comment (ambrinchaudhary) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js:58]: `handleSearchClick` sounds good!

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js:28]: Yeah, I think it makes sense, I'll change it!

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js (hunk 2 above)
Comment (ambrinchaudhary) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js:41]: `handleQueryChange` sound better to me, do you agree?

**Comment by ambrinchaudhary** [comment]
ci:stop

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Review by jaimelr10 — APPROVED** [review]
LGTM!

**Comment by ambrinchaudhary** [comment]
ci:forward

**Pull request closed without merging**